app/controllers/assignments_controller.rb

```diff
@@ -144,6 +150,7 @@ def index
 
   # Called on editing assignments (GET)
   def edit
+    authorize! with: AssignmentPolicy
```

Comment:
I think in this case you want to render a 403 error if this fails, not flash a message. Please handle this sort of authorization in a `before_action` context.